src/test/java/uk/nhs/digital/nhsconnect/lab/results/inbound/NhsAckProducerServiceTest.java

```diff
@@ -0,0 +1,237 @@
+import static org.mockito.Mockito.when;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import uk.nhs.digital.nhsconnect.lab.results.mesh.message.WorkflowId;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Interchange;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.InterchangeHeader;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.MessageHeader;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.InterchangeFactory;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.InterchangeParsingException;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.MessagesParsingException;
+import uk.nhs.digital.nhsconnect.lab.results.sequence.SequenceService;
+import uk.nhs.digital.nhsconnect.lab.results.utils.TimestampService;
+import uk.nhs.digital.nhsconnect.lab.results.inbound.MessageProcessingResult.Success;
+import uk.nhs.digital.nhsconnect.lab.results.inbound.MessageProcessingResult.Error;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+@ExtendWith(MockitoExtension.class)
+public class NhsAckProducerServiceTest {
+
+    private static final Instant FIXED_TIME = Instant.parse("2020-04-27T16:37:00Z");
+    private static final Long FIXED_SEQUENCE_NUMBER = 1000L;
+    private static final String NHSACK_IAF_RESPONSE_PATH = "/edifact/nhsAck_example_IAF.txt";
+    private static final String NHSACK_IAP_RESPONSE_PATH = "/edifact/nhsAck_example_IAP.txt";
+    private static final String NHSACK_IRA_RESPONSE_PATH = "/edifact/nhsAck_example_IRA.txt";
+    private static final String NHSACK_IRI_RESPONSE_PATH = "/edifact/nhsAck_example_IRI.txt";
+    private static final String NHSACK_IRM_RESPONSE_PATH = "/edifact/nhsAck_example_IRM.txt";
+    private static final String NHSACK_SCREENING_RESPONSE_PATH = "/edifact/nhsAck_screening_example.txt";
+
+    private static final String INTERCHANGE_SENDER = "000000004400001:80";
+    private static final String INTERCHANGE_RECIPIENT = "000000024600002:80";
+    private static final Long INTERCHANGE_CONTROL_REFERENCE = 1015L;
+    private static final Long MESSAGE_SEQUENCE_NUMBER_1 = 1L;
+    private static final Long MESSAGE_SEQUENCE_NUMBER_2 = 2L;
+    private static final Long MESSAGE_SEQUENCE_NUMBER_3 = 3L;
+
+
+    @InjectMocks
+    private NhsAckProducerService nhsAckProducerService;
+
+    @Mock
+    private TimestampService timestampService;
+
+    @Mock
+    private SequenceService sequenceService;
+
+    private final EdifactParser edifactParser = new EdifactParser(new InterchangeFactory());
```

Comment:
Could we reset this for each test to guarantee no leak between tests?

Reply:
Just to clarify: are you talking about SequenceService, EdifactParser or both?

Reply:
The `MockitoExtension` will take care of the `@Mock SequenceService` - I'm just talking about the EdifactParser. By reinitialising it for each test we can help to guarantee that one test won't interfere with the later ones.

Reply:
I've actually just removed it, it was leftover from another way of doing the tests. I'll try to keep that in mind in the future though!